core/process.py

```diff
@@ -1069,7 +1078,7 @@ def WhenDone(self, pid, status):
         self.state = job_state_e.Done
         if self.parent_pipeline:
             self.parent_pipeline.WhenDone(pid, status)
-        else:
+        elif self.position == POSITION_FOREGROUND:
```

Comment:
Hm thinking about my comment on the other PR

Can this be `elif not self.in_background` , and we assume that processes / pipelines are in the foreground except when run with `&`?

And then it changes state on STOPPED I guess?

It seems like we don't need 3 states?  2 should be enough?

Reply:
Yup. All this position stuff will be replaced with the things from #1675